packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,569 @@
+
+### POP/RPC
+
+`POP/RPC` works in an almost entirely similar way to `IFRAME/RPC`, except instead of rendering the `method` in an iframe, we render it in a popup. The same communication protocol between the rendered view and FCL applies:
+
+- A popup is rendered (comes from `endpoint` in the service).
+- The rendered popup says its ready `WalletUtils.sendMsgToFCL("FCL:VIEW:READY")`.
+- FCL will send the data to be dealt with `WalletUtils.sendMsgToFCL("FCL:VIEW:READY:RESPONSE", { ...body, service: {params, data} })`
+  - Where `body` is the stuff you care about, `params` and `data` are additional information you can provide in the service object.
+- The wallet sends back an Approved or Declined post message (It will be a `f_type: "PollingResponse"`, we will get to that in a bit)
+  - If it's approved, the polling responses data field will need to be what FCL is expecting.
+  - If it's declined, the polling responses reason field should say why it was declined.
+
+![POP/RPC Diagram](https://raw.githubusercontent.com/onflow/flow-js-sdk/master/packages/fcl/assets/service-method-diagrams/pop-rpc.png)
+
+### TAB/RPC
+
+`TAB/RPC` works in an almost entirely similar way to `IFRAME/RPC` and `POP/RPC`, except instead of rendering the `method` in an iframe or a popup, we render it in a new tab. The same communication protocol between the rendered view and FCL applies:
+
+- A new tab is rendered (comes from `endpoint` in the service).
+- The rendered tab says its ready `WalletUtils.sendMsgToFCL("FCL:VIEW:READY")`.
+- FCL will send the data to be dealt with `WalletUtils.sendMsgToFCL("FCL:VIEW:READY:RESPONSE", { ...body, service: {params, data} })`
+  - Where `body` is the stuff you care about, `params` and `data` are additional information you can provide in the service object.
+- The wallet sends back an Approved or Declined post message (It will be a `f_type: "PollingResponse"`, we will get to that in a bit)
+  - If it's approved, the polling responses data field will need to be what FCL is expecting.
+  - If it's declined, the polling responses reason field should say why it was declined.
+
+![TAB/RPC Diagram](https://raw.githubusercontent.com/onflow/flow-js-sdk/master/packages/fcl/assets/service-method-diagrams/tab-rpc.png)
+
+### HTTP/POST
+
+`HTTP/POST` initially sends a post request to the `endpoint` specified in the service, which should imediately return a `f_type: "PollingResponse"`.
+
+Like the `IFRAME/RPC` or `POP/RPS`, our goal is to eventually get an `APPROVED` or `DECLINED` polling response, and technically this endpoint could return one of those immediately.
+
+But more than likely that isn't the case and it will be in a `PENDING` state (`PENDING` is not available to `IFRAME/RPC` or `POP/RPC`).
+When the polling response is `PENDING` it requires an `update` field that includes a service, `BackChannelRpc`, that FCL can use to request an updated `PollingResponse` from.
```

Comment:
I think `update` needs to be `updates`, not 100% certain though can you check.

Reply:
It is `updates`, good catch